onnxruntime/contrib_ops/cpu/skip_layer_norm.cc

```diff
@@ -45,17 +45,16 @@ Status SkipLayerNorm<T>::Compute(OpKernelContext* p_ctx) const {
 
   const auto& input_dims = input->Shape().GetDims();
   size_t input_dims_size = input_dims.size();
+  const auto& skip_dims = skip->Shape().GetDims();
+  size_t skip_dims_size = skip_dims.size();
+
   if (input_dims_size != 3 && input_dims_size != 2) {
     return ORT_MAKE_STATUS(ONNXRUNTIME, INVALID_ARGUMENT,
                            "input is expected to have 3 or 2 dimensions, got ", input_dims_size);
   }
 
   int hidden_size = static_cast<int>(input_dims[input_dims_size - 1]);
 
-  if (input->Shape() != skip->Shape()) {
```

Comment:
Shall check that input and skip has same shape, or skip_dims[0] == 1 and can be broadcasted to input shape...

Reply:
Based on your recent code, I assume that we support the following cases:
(1)   input and skip: (batch_size, sequence_length, hidden_size)
(2)  input and skip: (batch_size * sequence_length, hidden_size)
(3)   input: (batch_size, sequence_length, hidden_size), skip: (1, sequence_length, hidden_size)
(4)  input: (batch_size, sequence_length, hidden_size), skip: (sequence_length, hidden_size)

Is my understanding correct?

Reply:
Yes, this PR adds (3) and (4)